src/scrapers/leumi.js

```diff
@@ -168,22 +187,120 @@ async function fetchTransactionsForAccount(page, startDate) {
+      const txn = {
+        date: txnColumns[0],
+        description: txnColumns[1],
+        reference: txnColumns[2],
+        debit: txnColumns[3],
+        credit: txnColumns[4],
+        status: TRANSACTION_STATUS.COMPLETED,
+      };
+      txns.push(txn);
+    }
+  }
+
+  return txns;
+}
+
+async function fetchForeignAccountsList(page) {
+  const accounts = [];
+  const accountRows = await pageEvalAll(page, 'table#ctlForeignAccounts tr.item');
+
+  for (let accountRowIndex = 0; accountRowIndex < accountRows.length; accountRowIndex += 1) {
+    const accountUrl = await pageEval(accountRows[accountRowIndex], 'td.ForeignColumn1 a', (anchor) => {
+      return anchor.getAttribute('href');
+    });
+
+    const accountCurrencyLabel = await pageEval(accountRows[accountRowIndex], 'td.ForeignColumn2 span', (span) => {
+      return span.innerText;
+    });
+
+    let currency = null;
+    switch (accountCurrencyLabel) {
+      case DOLLAR_CURRENCY_LABEL:
+        currency = DOLLAR_CURRENCY;
+        break;
+      default:
+        throw new Error(`failed to extract foreign account transactions, unknown currency '${accountCurrencyLabel}'`);
+    }
+
+    if (accountUrl) {
+      accounts.push({
+        url: accountUrl,
+        currency,
+      });
+    }
+  }
+
+  return accounts;
+}
+
+async function fetchTransactionsForForeignAccounts(page, startDate) {
+  const result = [];
+  const url = `${BASE_URL}/ebanking/Accounts/ExtendedSummary.aspx?DisplayType=2&from=sideMenu`;
+  await navigateTo(page, url);
+
+  const selectedSnifAccount = await page.$eval('#ddlClientNumber_m_ddl option[selected="selected"]', (option) => {
+    return option.innerText;
+  });
+
+  const accountNumber = selectedSnifAccount.replace('/', '_');
+  const accounts = await fetchForeignAccountsList(page);
+
+  for (let accountIndex = 0; accountIndex < accounts.length; accountIndex += 1) {
+    const account = accounts[accountIndex];
+    await navigateTo(page, `${BASE_URL}${account.url}`);
+
+    await dropdownSelect(page, 'select#ddlTransactionPeriod', '3');
+    await waitUntilElementFound(page, 'select#ddlTransactionPeriod');
+    await fillInput(
+      page,
+      'input#dtFromDate_textBox',
+      startDate.format(DATE_FORMAT),
+    );
+    await clickButton(page, 'input#btnDisplayDates');
+    await waitForNavigation(page);
+    await waitUntilElementFound(page, 'div#WorkSpaceBox > table:first-child');
+
+    const txns = await fetchCompletedTransactionsForForeignAccount(page);
+
+    result.push({
+      accountNumber,
+      txns: convertTransactions(txns, account.currency),
+    });
+  }
+
+  return result;
+}
+
 async function fetchTransactions(page, startDate) {
-  // TODO need to extend to support multiple accounts and foreign accounts
-  return [await fetchTransactionsForAccount(page, startDate)];
+  return [
+    await fetchTransactionsForLocalAccount(page, startDate),
+    ...await fetchTransactionsForForeignAccounts(page, startDate),
```

Comment:
@esakal mind extracting the `await` calls into variables and then do `[localTxns, ...foreignTxs]`? Would be cleaner.